### PR TITLE
Copy empty folders and files (#28)

### PR DESCRIPTION
**Pull request by joehand**
Fixes empty file and directory bug (#28)

I added one test for the empty file. I was having trouble testing the empty directory issue though - can you run `appendFile` on a directory somehow?

### CODE REVIEW

**Comment by mafintosh** [comment]
Yea appendFile works with directories as well (which makes the name a bit silly)

File: index.js [hunk 1]
```diff
@@ -156,7 +156,22 @@ Archive.prototype.download = function (i, cb) {
...
+
+    if (!feed) return createEmtpyEntry()
+
+    function createEmtpyEntry () {
+      var dir = dest
+      if (entry.type === 'file') dir = path.dirname(dest)
+      mkdirp(dir, function (err) {
+        if (err) return cb(err)
+        if (entry.type !== 'file') return cb(null)
+        fs.open(dest, 'a', function (err) {
```
Comment (mafintosh) [inline index.js]: fs.open calls the callback with a file descriptor as well. you should close that afterwards

```
fs.open(..., function (err, fd) {
  if (err) return cb(err)
  fs.close(fd, cb)
})
```

File: index.js [hunk 2]
```diff
@@ -156,7 +156,22 @@ Archive.prototype.download = function (i, cb) {
     if (err) return cb(err)
 
     var feed = self._getFeed(entry)
-    if (!feed) return cb(null)
+    var dest = join(self.directory, entry.name)
+
+    if (!feed) return createEmtpyEntry()
+
+    function createEmtpyEntry () {
```
Comment (mafintosh) [inline index.js]: small nit pick. could you move this to the btm of the function with the other funcs?

Comment (mafintosh) [inline index.js]: also i think you typo'ed the name hehe

**Comment by mafintosh** [comment]
Looking really good!

File: index.js (hunk 2 above)
Comment (joehand) [inline index.js]: Sure, will move down. haha thanks for catching that.

**Comment by mafintosh** [comment]
3.3.1

**Pull request merged**